python/dgl/backend/pytorch/tensor.py

```diff
@@ -349,6 +349,10 @@ def zerocopy_to_dgl_ndarray(data):
         return nd.from_dlpack(dlpack.to_dlpack(data.contiguous()))
 
 def zerocopy_to_dgl_ndarray_for_write(input):
+    assert input.is_contiguous(), "Cannot convert non-contiguous tensors " \
+        "to dgl ndarray for write."
+    assert input.numel() == input.storage().size(), "Cannot convert view " \
```

Comment:
I doubt the error message here would confuse users. For example in this issue https://github.com/dmlc/dgl/issues/3939, users would question: "I just want to use view tensors as input, not for write. Why will it throw a 'not for write' error?"

Reply:
I wrapped the calls from dgl.utils with `try`/`except`, such that the user will be told pinning the memory failed in e279be8.